src/app/modules/MainScreen/index.tsx

```diff
@@ -1,7 +1,39 @@
+import { compose, bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { injectIntl } from 'react-intl';
+
+import {
+  incrementValue,
+  decrementValue,
+  incrementAsyncValue,
+} from './store/actions';
+
 import MainScreen from './MainScreen';
 
-export default {
-  module: MainScreen,
+const mapStateToProps = (state: any): any => {
+  return {
+    value: state.ageValue.value,
+  };
+};
+
+const mapDispatchToProps = (dispatch: any): any => {
+  return bindActionCreators(
+    {
+      incrementValue,
+      decrementValue,
+      incrementAsyncValue,
+    },
+    dispatch,
+  );
+};
+
+const MainScreenComposed = compose(
```

Comment:
what about having something like this:

```
import { NavigationScreenComponent } from 'react-navigation';
import { NavigationStackOptions } from 'react-navigation-stack';

....

const MainScreenComposed = compose<NavigationScreenComponent<any, any>>(
  injectIntl,
  connect(mapStateToProps, mapDispatchToProps),
)(MainScreen);

interface ScreenPropsConfig {
  options: NavigationStackOptions;
  module: NavigationScreenComponent<any, any>;
  name: string;
}

export const MainScreenModule: ScreenPropsConfig = {
  module: MainScreenComposed,
  name: 'MainScreen',
  options: {
    headerLeft: null,
    headerTitle: null,
    headerRight: null,
  },
};
```

`ScreenPropsConfig` interface would be a common type used for all modules.

@grzegorzkruk @lukaszblasz ?

Reply:
Im fine with that, but it shoudl not be placed there, So ScreenPropsConfig should be put in ru domain/common typing and imported here and also in every module in application.